### PR TITLE
WhatsApp template language support

### PR DESCRIPTION
**Pull request by jerith**
## Purpose
Hardcoding `en_US` for all WhatsApp templates isn't ideal for non-English content. We should use the relevant locale instead.

## Approach
Look up the closest WhatsApp supported language code for the page locale and use that instead of a hardcoded `en_US`.

### CODE REVIEW

**Comment by Buhle79** [comment]
Hi @jerith I've resolved the conflicts and merged with main.

File: home/tests/test_whatsapp.py [hunk 1]
```diff
@@ -166,10 +187,47 @@ def test_create_whatsapp_template_with_image(self, tmp_path, settings):
...
+    def test_create_whatsapp_template_with_language(self) -> None:
+        """
+        When we create a WhatsApp template, the language is converted to the
+        appropriate WhatsApp language code.
+        """
+        url = "http://whatsapp/graph/v14.0/27121231234/message_templates"
+        responses.add(responses.POST, url, json={})
+
+        # The default English locale gives us a language of "en_US".
+        # FIXME: Should this be "en" instead?
```
Comment (rudigiesler) [inline home/tests/test_whatsapp.py:203]: It looks like we can define both in our settings file: https://docs.wagtail.org/en/stable/reference/settings.html#wagtail-content-languages

Maybe we should ensure that that list matches what we can do with WhatsApp templates? We probably don't want to remove any languages from the list, but we can do something like have all 3 of `en`, `en-us`, `en-gb`, and then for backwards compatibility add a migration that changes any `en` locales to `en-us`

File: home/whatsapp.py [hunk 2]
```diff
@@ -1,15 +1,121 @@
...
+    category: str,
+    locale: Locale | None = None,
+    quick_replies: Iterable[str] = (),
+    image_id: int | None = None,
+    example_values: Iterable[str] | None = None,
+) -> None:
+    """
+    Create a WhatsApp template through the WhatsApp Business API.
+
+    FIXME: Do we want locale to be optional?
```
Comment (rudigiesler) [inline home/whatsapp.py:114]: The language field is required for the template API, so I would say that we want locale to be required here too.

**Review by rudigiesler — COMMENTED** [review]
Trying to answer all the TODOs and fixme's here. I think I've come up with a way of supporting all 3 of `en`, `en-us`, `en-gb`, while keeping backwards compat, let me know what you think.

File: home/tests/test_whatsapp.py (hunk 1 above)
Comment (jerith) [reply home/tests/test_whatsapp.py:203]: This (somewhat poorly worded, sorry) FIXME was about switching the default WhatsApp language for wagtail's `en` from the current `en_US` to `en`. I think we've already decided to keep that `en_US` for backwards compatibility with existing templates.

File: home/tests/test_models.py [hunk 3]
```diff
@@ -79,11 +81,12 @@ def test_template_create_on_save_deactivated(self, mock_create_whatsapp_template
     @mock.patch("home.models.create_whatsapp_template")
     def test_template_create_on_save(self, mock_create_whatsapp_template):
         page = create_page(is_whatsapp_template=True)
+        locale = Locale.objects.get(language_code="en")
```
Comment (jerith) [inline home/tests/test_models.py:84]: Can we call this `en` rather than `locale`, please?

File: home/tests/test_models.py [hunk 4]
```diff
@@ -358,6 +368,25 @@ def test_for_missing_migrations(self):
             "There are missing migrations:\n %s" % output.getvalue(),
         )
 
+    @override_settings(WHATSAPP_CREATE_TEMPLATES=True)
+    @mock.patch("home.models.create_whatsapp_template")
+    @pytest.mark.xfail(
+        reason="This fails because we can't get locale to create the page, "
+        "these tests will be changed once whatsapp templates are separated."
+    )
```
Comment (jerith) [inline home/tests/test_models.py:376]: Note for other reviewers: We're replacing these tests with better ones (that don't use mocks, to start with) in #277. It's not worth duplicating that work here.

**Review by fritzbrand — APPROVED** [review]
Looks good to me

**Pull request merged**